core/trino-main/src/test/java/io/trino/sql/planner/assertions/TableFunctionProcessorMatcher.java

```diff
@@ -142,6 +145,10 @@ public MatchResult detailMatches(PlanNode node, StatsProvider stats, Session ses
             }
         }
 
+        if (!hashSymbol.map(symbolAliases::get).equals(tableFunctionProcessorNode.getHashSymbol().map(Symbol::toSymbolReference))) {
```

Comment:
This enforces checking for hash symbol if `tableFunctionProcessorNode` is to return one. Check will not be skipped if `TableFunctionProcessorMatcher.hashSymbol` is not set. Intentional?

Reply:
fixed.